### PR TITLE
33712 restrict tool names

### PR DESCRIPTION
**Pull request by IssacAX123**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

#33712

### CODE REVIEW

File: dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpCdiExtension.java [hunk 1]
```diff
@@ -225,6 +226,26 @@ private boolean reportOnDuplicateTools(AfterDeploymentValidation afterDeployment
...
+        boolean error = false;
+        for (ToolMetadata tool : tools.getAllTools()) {
+            List<String> errors = new ArrayList<>();
+            String toolName = tool.name();
+            if (toolName.length() == 0 || toolName.length() > 128) {
+                errors.add("Tool names should be between 1 and 128 characters in length (inclusive).");
+            }
+            if (!toolName.matches("[\\d\\w.-]+")) {
+                errors.add("The following should be the only allowed characters: uppercase and lowercase ASCII letters (A-Z, a-z), digits (0-9), underscore (_), hyphen (-), and dot (.). Tool names should not contain spaces, commas, or other special characters.");
+            }
```
Comment (Azquelt) [inline dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpCdiExtension.java:216]: These strings also need to be translated, so it would make sense to include them in the message rather than adding them on the end. We've only got two cases, two similar messages here would be fine.

File: dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops [hunk 2]
```diff
@@ -194,3 +195,14 @@ CWMCM0020E.defaultvalue.conversion.error.useraction=Ensure that the default valu
...
+
+
+# Used for server logging
+# do not translate MCP
+# {1} = The qualified name of the tool
+# {1} = The name of the tool
+# {2} = The list of MCP spec naming conventions broken
+CWMCM0022E.invalid.tool.name=CWMCM0022E: MCP tool method {0} has the name {1} which is invalid as it does not follow the following MCP naming conventions: {2}
+CWMCM0022E.invalid.tool.name.explanation=One MCP Tool name dosent follow the MCP naming convention for tools.
+CWMCM0022E.invalid.tool.name.useraction=Provide a tool name that follows the MCP naming convention.
```
Comment (Azquelt) [inline dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops]: Message needs work, check the guidelines. The placeholder comments are also incorrect.

File: dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/deploymentErrorApps/InvalidToolNameTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        return arg;
+    }
+
+    @Tool(name = "")
+    public String invalidTool4(@ToolArg(name = "arg") String arg) {
+        return arg;
+    }
+
+    @Tool(name = "ibmopenlibertyibmopenlibertyibmopenlibertyibmopenliberty ibmopenlibertyibmopenlibertyibmopenlibertyibmopenliberty_ibmopenlibertyibmopenliberty")
+    public String invalidTool5(@ToolArg(name = "arg") String arg) {
```
Comment (Azquelt) [inline dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/deploymentErrorApps/InvalidToolNameTest.java]: remove `ibm` here

File: dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpCdiExtension.java [hunk 4]
```diff
@@ -225,6 +226,26 @@ private boolean reportOnDuplicateTools(AfterDeploymentValidation afterDeployment
...
 
+    private boolean reportOnInvalidToolNames(AfterDeploymentValidation afterDeploymentValidation) {
+        boolean error = false;
+        for (ToolMetadata tool : tools.getAllTools()) {
+            List<String> errors = new ArrayList<>();
+            String toolName = tool.name();
+            if (toolName.length() == 0 || toolName.length() > 128) {
+                errors.add("Tool names should be between 1 and 128 characters in length (inclusive).");
+            }
+            if (!toolName.matches("[\\d\\w.-]+")) {
```
Comment (Azquelt) [inline dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpCdiExtension.java]: - `\w` includes `\d` so you don't need both
- this could be a precompiled regex (use `Pattern.compile` to make a constant, then use `Pattern.matches` to match against the tool name)
- might even want to split the check out into its own method so it can be unit tested
  - We should have tests that we don't accept non-ascii letters and digits like é, ᴀ, ぁ, 𝟏, or similar punctuation like —. I think the current code is correct, `\w` would only match those in unicode mode, but it would be good to test these edge cases.

Side note: it's kind of weird that `\\w` retains its character class meaning within `[]` while `.` does not, but this is correct.

File: dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/DeploymentProblemTest.java [hunk 5]
```diff
@@ -102,23 +104,72 @@ public void testGenericArgsTestCase() throws Exception {
...
+//        List<String> expectedErrorList = List.of("io.openliberty.mcp.internal.fat.tool.deploymentErrorApps.InvalidToolNameTest.invalidTool4");
+//        ExpectedAppFailureValidator.findAndAssertExpectedErrorsInLogs("Invalid Tool name: ", expectedErrorHeader, expectedErrorList, server);
+//    }
+//
+//    @Test
+//    public void testInvalidToolNamesWithinvalidLengthandCharactersTestCase() throws Exception {
+//        String expectedErrorHeader = "MCP tool method named (.+?) is an invalid name as it does not follow the following MCP naming conventions: (.+?)";
+//        List<String> expectedErrorList = List.of("io.openliberty.mcp.internal.fat.tool.deploymentErrorApps.InvalidToolNameTest.invalidTool5");
+//        ExpectedAppFailureValidator.findAndAssertExpectedErrorsInLogs("Invalid Tool name: ", expectedErrorHeader, expectedErrorList, server);
+//    }
```
Comment (Azquelt) [inline dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/DeploymentProblemTest.java]: Remove commented out tests.

**Review by Azquelt — CHANGES_REQUESTED** [review]

File: dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops [hunk 6]
```diff
@@ -201,3 +202,20 @@ CWMCM0021E.defaultvalue.character.invalid.length.useraction=Ensure that the defa
 CWMCM0022E.invalid.cursor.value=CWMCM0022E: The "{0}" cursor value is invalid.
 CWMCM0022E.invalid.cursor.value.explanation=A cursor value was provided as a parameter to the tool/list command, but the value is invalid.
 CWMCM0022E.invalid.cursor.value.useraction=Ensure that the cursor value is provided exactly as it was returned from the previous tool/list call.
+
+# Used for server logging
+# do not translate MCP
+# {0} = The qualified name of the tool
+# {1} = The name of the tool
+CWMCM0023E.invalid.length.tool.name=CWMCM0023E: The MCP tool method {0} has the name {1} which is invalid. Tool names must be between 1 and 128 characters in length inclusive.
```
Comment (Azquelt) [inline dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops]: This looks much better. I _think_ ID will ask for this change:

```suggestion
CWMCM0023E.invalid.length.tool.name=CWMCM0023E: The {1} name of the {0} MCP tool method is invalid. Tool names must be between 1 and 128 characters in length inclusive.
```

Same for the other message.

File: dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpCdiExtensionMetadata.java [hunk 7]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2025 IBM Corporation and others.
+ * Copyright (c) 2025, 2026 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
```
Comment (Azquelt) [inline dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpCdiExtensionMetadata.java:3]: Confused what's happened with this class. We're changing the copyright date but have no changes to the file.

File: dev/io.openliberty.mcp.internal/test/io/openliberty/mcp/internal/test/McpCdiExtentionTest.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+package io.openliberty.mcp.internal.test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import io.openliberty.mcp.internal.McpCdiExtension;
+
+public class McpCdiExtentionTest {
```
Comment (Azquelt) [inline dev/io.openliberty.mcp.internal/test/io/openliberty/mcp/internal/test/McpCdiExtentionTest.java]:
```suggestion
public class McpCdiExtensionTest {
```

**Review by Azquelt — CHANGES_REQUESTED** [review]

**Comment by IssacAX123** [comment]
@OpenLiberty/message-reviewer
Hi, please could you review CWMCM0024E.invalid.character.tool.name and CWMCM0023E.invalid.length.tool.name in [CWMCM.nlsprops](https://github.com/OpenLiberty/open-liberty/blob/1b52841e9c2e3c9c52f0173d13adc4f719e497e3/dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops)

File: dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops [hunk 9]
```diff
@@ -201,3 +202,20 @@ CWMCM0021E.defaultvalue.character.invalid.length.useraction=Ensure that the defa
 CWMCM0022E.invalid.cursor.value=CWMCM0022E: The "{0}" cursor value is invalid.
 CWMCM0022E.invalid.cursor.value.explanation=A cursor value was provided as a parameter to the tool/list command, but the value is invalid.
 CWMCM0022E.invalid.cursor.value.useraction=Ensure that the cursor value is provided exactly as it was returned from the previous tool/list call.
+
+# Used for server logging
+# do not translate MCP
+# {0} = The name of the tool 
+# {1} = The qualified name of the tool
+CWMCM0023E.invalid.length.tool.name=CWMCM0023E: The {0} name of the {1} MCP tool method is invalid. Tool names must be between 1 and 128 characters in length inclusive.
```
Comment (chirp1) [inline dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops]: Change the message text to:
The {0} tool name for the {1} MCP tool method is not valid. Tool names must be between 1 and 128 characters in length, inclusive.

File: dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops [hunk 10]
```diff
@@ -201,3 +202,20 @@ CWMCM0021E.defaultvalue.character.invalid.length.useraction=Ensure that the defa
...
+CWMCM0023E.invalid.length.tool.name.explanation=The tool name dosent follow the MCP specification of names being between 1 and 128 characters inclusive.
```
Comment (chirp1) [inline dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops]: Change the wording to:

The tool name doesn't follow the MCP specification for names. Names must be between 1 and 128 characters in length, inclusive.

File: dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops [hunk 11]
```diff
@@ -201,3 +202,20 @@ CWMCM0021E.defaultvalue.character.invalid.length.useraction=Ensure that the defa
...
+CWMCM0023E.invalid.length.tool.name.useraction=Ensure tool names between 1 and 128 characters inclusive.
```
Comment (chirp1) [inline dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops]: Change the wording to:

Make sure that tool names are between 1 and 128 characters, inclusive.

File: dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops [hunk 12]
```diff
@@ -201,3 +202,20 @@ CWMCM0021E.defaultvalue.character.invalid.length.useraction=Ensure that the defa
...
+
+# Used for server logging
+# do not translate MCP
+# {0} = The name of the tool 
+# {1} = The qualified name of the tool
+CWMCM0024E.invalid.character.tool.name=CWMCM0024E: The {0} name of the {1} MCP tool method is invalid. Tool names must only contain ASCII letters, digits, underscores or hyphens.
```
Comment (chirp1) [inline dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops]: Change the message text to:

The {0} tool name for the {1} MCP tool method is not valid. Tool names must contain only ASCII letters, digits, underscores, or hyphens.

File: dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops [hunk 13]
```diff
@@ -201,3 +202,20 @@ CWMCM0021E.defaultvalue.character.invalid.length.useraction=Ensure that the defa
...
+CWMCM0024E.invalid.character.tool.name.explanation=The tool name dosent follow the MCP specification of names only containing ASCII letters, digits, underscores, hyphens and not any other special character.
```
Comment (chirp1) [inline dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops]: Change the wording to:

The tool name doesn't follow the MCP specification for names. Names can contain only ASCII letters, digits, underscores, hyphens, and not any other special character.

File: dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops [hunk 14]
```diff
@@ -201,3 +202,20 @@ CWMCM0021E.defaultvalue.character.invalid.length.useraction=Ensure that the defa
...
+CWMCM0024E.invalid.character.tool.name.useraction=Ensure tool names contain only ASCII letters, digits, underscores or hyphens.
```
Comment (chirp1) [inline dev/io.openliberty.mcp.internal/resources/io/openliberty/mcp/internal/resources/CWMCM.nlsprops]: Change the wording to:

Make sure that tool names contain only ASCII letters, digits, underscores, or hyphens.

**Review by chirp1 — COMMENTED** [review]
I reviewed the messages and provided my comments.

**Review by Azquelt — APPROVED** [review]
Looks good, thanks
